pkg/ibgu/ibgu.go

```diff
@@ -0,0 +1,331 @@
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	if len(actions) == 0 {
+		glog.V(100).Info("The 'actions' slice is empty")
+
+		builder.errorMsg = "Plan actions cannot be empty"
+
+		return builder
+	}
+
+	if maxConcurrency <= 0 {
+		glog.V(100).Infof("Invalid maxConcurrency value: %d", maxConcurrency)
+
+		builder.errorMsg = "maxConcurrency must be greater than 0"
+
+		return builder
+	}
+
+	if timeout <= 0 {
+		glog.V(100).Infof("Invalid timeout value: %d", timeout)
+
+		builder.errorMsg = "timeout must be greater than 0"
+
+		return builder
+	}
+
+	plan := v1alpha1.PlanItem{
+		Actions: actions,
+		RolloutStrategy: v1alpha1.RolloutStrategy{
+			MaxConcurrency: maxConcurrency,
+			Timeout:        timeout,
+		},
+	}
+
+	builder.Definition.Spec.Plan = append(builder.Definition.Spec.Plan, plan)
+
+	return builder
+}
+
+// Get returns imagebasedgroupupgrade object if found.
+func (builder *IbguBuilder) Get() (*v1alpha1.ImageBasedGroupUpgrade, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting imagebasedgroupupgrade %s",
+		builder.Definition.Name)
+
+	imagebasedgroupupgrade := &v1alpha1.ImageBasedGroupUpgrade{}
+
+	err := builder.apiClient.Get(context.TODO(), goclient.ObjectKey{
+		Name:      builder.Definition.Name,
+		Namespace: builder.Definition.Namespace,
+	}, imagebasedgroupupgrade)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return imagebasedgroupupgrade, err
+}
+
+// Exists checks whether the given imagebasedgroupupgrade exists.
+func (builder *IbguBuilder) Exists() bool {
+	if valid, _ := builder.validate(); !valid {
+		return false
+	}
+
+	glog.V(100).Infof("Checking if imagebasedupgrade %s exists",
+		builder.Definition.Name)
+
+	var err error
+	builder.Object, err = builder.Get()
+
+	return err == nil || !k8serrors.IsNotFound(err)
+}
+
+// Create makes an IBGU in the cluster and stores the created object in struct.
+func (builder *IbguBuilder) Create() (*IbguBuilder, error) {
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	glog.V(100).Infof("Creating the imageasedgroupupgrade %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	var err error
+	if !builder.Exists() {
+		err = builder.apiClient.Create(context.TODO(), builder.Definition)
+		if err == nil {
+			builder.Object = builder.Definition
+		}
+	}
+
+	return builder, err
+}
+
+// Delete removes the IBGU from the cluster.
+func (builder *IbguBuilder) Delete() error {
+	if valid, err := builder.validate(); !valid {
+		return err
+	}
+
+	glog.V(100).Infof("Deleting the ImageBasedGroupUpgrade %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if builder.Object == nil {
+		builder.Object, _ = builder.Get()
+	}
+
+	if builder.Object != nil {
+		err := builder.apiClient.Delete(context.TODO(), builder.Object)
+		if err != nil {
+			return err
+		}
+
+		builder.Object = nil
+	}
```

Comment:
It's equivalent so not a big deal but I think using `builder.Exists()` like in [other delete functions](https://github.com/openshift-kni/eco-goinfra/blob/main/pkg/deployment/deployment.go#L430-L443) should be preferred for consistency

Reply:
Sure, updated. Tnx for the pointer!